src/utils/venvUtils.ts

```diff
@@ -3,57 +3,106 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as fse from 'fs-extra';
 import * as path from 'path';
-import { workspace, WorkspaceConfiguration, WorkspaceFolder } from 'vscode';
-import { Platform } from "../constants";
+import { workspace, WorkspaceConfiguration } from 'vscode';
+import { Platform, pythonVenvSetting } from "../constants";
 import { ext } from '../extensionVariables';
 import { getFuncExtensionSetting } from '../ProjectSettings';
 import { cpUtils } from './cpUtils';
 
 export namespace venvUtils {
-    const bashAndCmdSeparator: string = ' && ';
-    const powerShellSeparator: string = ' ; ';
+    enum Terminal {
+        bash,
+        cmd,
+        powerShell
+    }
 
-    export function convertToVenvTask(folder: WorkspaceFolder, ...commands: string[]): string {
-        const venvName: string | undefined = getFuncExtensionSetting<string>('pythonVenv', folder.uri.fsPath);
+    export function convertToVenvCommand(command: string, folderPath: string, platform: NodeJS.Platform = process.platform): string {
+        const terminal: Terminal = getTerminal(platform);
+        const venvName: string | undefined = getFuncExtensionSetting<string>(pythonVenvSetting, folderPath);
         if (venvName) {
-            commands.unshift(getVenvActivateCommand(venvName));
+            return joinCommands(terminal, getVenvActivateCommand(venvName, terminal, platform), command);
+        } else {
+            return command;
```

Comment:
Should you silently return the command without the venv activation?  Won't this break users?

Reply:
It's not our requirement that they use a venv, it's the func cli's requirement. I'm not sure if they'll always have that requirement so I don't want to hard code an error into our code.